language/move-vm/runtime/src/unit_tests/arguments.rs

```diff
@@ -0,0 +1,439 @@
+        SignatureToken::Reference(Box::new(SignatureToken::Signer)),
+        SignatureToken::Bool,
+        SignatureToken::Address,
+    ]));
+    call_script(
+        script,
+        vec![Value::bool(false), Value::address(AccountAddress::random())],
+    )
+    .expect("good signature");
+    let script = make_script(Signature(vec![
+        SignatureToken::Bool,
+        SignatureToken::Vector(Box::new(SignatureToken::U8)),
+        SignatureToken::Vector(Box::new(SignatureToken::Vector(Box::new(
+            SignatureToken::Address,
+        )))),
+    ]));
+    let mut addresses = vec![];
+    addresses.push(Value::vector_address(vec![
+        AccountAddress::random(),
+        AccountAddress::random(),
+    ]));
+    addresses.push(Value::vector_address(vec![
+        AccountAddress::random(),
+        AccountAddress::random(),
+    ]));
+    addresses.push(Value::vector_address(vec![
+        AccountAddress::random(),
+        AccountAddress::random(),
+    ]));
+    let values = Value::constant_vector_generic(
+        addresses,
+        &SignatureToken::Vector(Box::new(SignatureToken::Address)),
+    )
+    .expect("build vector<vector<address>>");
+    call_script(
+        script,
+        vec![Value::bool(true), Value::vector_u8(vec![0, 1]), values],
+    )
+    .expect("good signature");
+}
+
+#[test]
+fn check_constant_args() {
+    //
+    // Simple arguments
+    //
+
+    // error: wrong arg
+    let script = make_script(Signature(vec![SignatureToken::U128]));
+    call_script(script, vec![Value::u128(0)]).expect("good u128 failed");
+
+    // error: no args
+    let script = make_script(Signature(vec![SignatureToken::U64]));
+    call_script(script, vec![]).expect_err("missing arguments");
+
+    // error: too many args
+    let script = make_script(Signature(vec![SignatureToken::Bool]));
+    call_script(script, vec![Value::bool(true), Value::u8(0)]).expect_err("too many arguments");
+
+    //
+    // Vector arguments
+    //
+
+    // success: vector of addresses
+    let script = make_script(Signature(vec![SignatureToken::Vector(Box::new(
+        SignatureToken::Address,
+    ))]));
+    // empty vector
+    call_script(script.clone(), vec![Value::vector_address(vec![])])
+        .expect("good vector<address> failed");
+    // one elem vector
+    call_script(
+        script.clone(),
+        vec![Value::vector_address(vec![AccountAddress::random()])],
+    )
+    .expect("good vector<address> failed");
+    // multiple elems vector
+    call_script(
+        script.clone(),
+        vec![Value::vector_address(vec![
+            AccountAddress::random(),
+            AccountAddress::random(),
+            AccountAddress::random(),
+            AccountAddress::random(),
+            AccountAddress::random(),
+        ])],
+    )
+    .expect("good vector<address> failed");
+    // wrong vector
+    call_script(script.clone(), vec![Value::vector_bool(vec![true])])
+        .expect_err("bad vector<address>");
```

Comment:
Should we be matching on the status here for these? or do we not care as long as it failed?